File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/download/DiagnosisKeysDownloadResponse.java
package uk.nhs.nhsx.keyfederation.download;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class DiagnosisKeysDownloadResponse {

    public final String batchTag;
    public final List<ExposureDownload> exposures;

    @JsonCreator
    public DiagnosisKeysDownloadResponse(@JsonProperty String batchTag,
                                         @JsonProperty List<ExposureDownload> exposures) {
        this.batchTag = batchTag;
        this.exposures = exposures;
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/keyfederation/download/DiagnosisKeysDownloadResponse.kt
package uk.nhs.nhsx.keyfederation.download

import uk.nhs.nhsx.keyfederation.BatchTag

sealed class InteropDownloadResponse
data class DiagnosisKeysDownloadResponse(val batchTag: BatchTag, val exposures: List<ExposureDownload>): InteropDownloadResponse()
object NoContent: InteropDownloadResponse()
